### PR TITLE
Added custom generateKey method server.action method options

### PR DESCRIPTION
**Pull request by ooogway**
1. Added custom generateKey method server.action method options
2. Restricted Hapi version to < 9.4.2
3. Updated README for custom generateKey method
4. Updated package version 0.3.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,10 +98,22 @@ server.register(Chairo, function (err) {
...
+        return next(null, { average: ( message.samples.dataset.values[0] + message.samples.dataset.values[0] ) / 2 });
+    });
+
 	// Map action to a hapi server method
 
     server.action('generate', 'generate:id', { cache: { expiresIn: 1000, generateTimeout: 3000 } });
 
+    // Map action to a hapi server method with custom generateKey method
+
+    server.action('average', 'calc:average', { cache: { expiresIn: 1000, generateTimeout: 3000 } }, generateKey: function(message){
```
Comment (geek) [inline README.md]: add a couple of spaces on the function:

```
function (message) {
```

File: package.json [hunk 2]
```diff
@@ -23,12 +23,12 @@
...
   },
   "peerDependencies": {
-    "hapi": ">=8.5.0"
+    "hapi": "8.5.0 - 9.4.2"
   },
   "devDependencies": {
     "code": "1.x.x",
     "handlebars": "4.x.x",
-    "hapi": ">=8.5.0",
+    "hapi": "8.5.0 - 9.4.2",
```
Comment (geek) [inline package.json]: should be safe to allow any patch version for 9:

```
"hapi": "8.5.0 - 9.4.x",
```

will that work?

File: test/index.js [hunk 3]
```diff
@@ -109,6 +109,40 @@ describe('action()', function () {
...
+        server.register({ register: Chairo, options: { log: 'silent' } }, function (err) {
+
+            expect(err).to.not.exist();
+
+            server.seneca.add({ generate: 'id' }, function (message, next) {
+
+                return next(null, { result: message.samples.readings.values[0] * message.samples.readings.values[1] });
+            });
+
+            server.action('generate', { generate:'id' }, { cache: { expiresIn: 1000, generateTimeout: 3000 }, generateKey: function (message) {
```
Comment (geek) [inline test/index.js]:
```
generate: 'id'
```

File: test/index.js [hunk 4]
```diff
@@ -109,6 +109,40 @@ describe('action()', function () {
...
+                return 'id' +  message.samples.readings.values[0] + ':' +  message.samples.readings.values[1];
+            } });
+
+            server.start(function () {
+
+                server.methods.generate({ samples: { readings: { values: [2, 3] } } }, function (err, result1) {
+
+                    expect(result1).to.deep.equal({ result: 6 });
+
+                    server.methods.generate({ samples: { readings: { values: [2,3] } } }, function (err, result2) {
```
Comment (geek) [inline test/index.js]:
```
[2, 3]
```

File: test/index.js [hunk 5]
```diff
@@ -109,6 +109,40 @@ describe('action()', function () {
...
+            server.action('generate', { generate:'id' }, { cache: { expiresIn: 1000, generateTimeout: 3000 }, generateKey: function (message) {
+
+                return 'id' +  message.samples.readings.values[0] + ':' +  message.samples.readings.values[1];
+            } });
+
+            server.start(function () {
+
+                server.methods.generate({ samples: { readings: { values: [2, 3] } } }, function (err, result1) {
+
+                    expect(result1).to.deep.equal({ result: 6 });
```
Comment (geek) [inline test/index.js]: toss in a couple of error checks:

```
expect(err).to.not.exist();
```

**Comment by ooogway** [comment]
will submit changes in 30. 

should I also submit a Hapi v > 10 version, with version bump to 1.0.0 ?

**Comment by geek** [comment]
@ooogway we can do hapi 10 in a next release/separate PR...

File: package.json (hunk 2 above)
Comment (ooogway) [inline package.json]: 9.x.x or 9.4.x ?

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "chairo",
   "description": "Seneca micro-services plugin for hapi",
-  "version": "0.2.0",
+  "version": "0.3.0",
```
Comment (ooogway) [inline package.json]: 0.2.1 right, instead of 0.3.0 ?

File: package.json (hunk 2 above)
Comment (geek) [inline package.json]: 9.x.x should be safe

File: package.json (hunk 6 above)
Comment (geek) [inline package.json]: I will update the version when I release, so you can keep it unchanged.

**Comment by ooogway** [comment]
@geek changes made. please review.

**Pull request merged**